docs/pages/machine-id/guides/github-actions.mdx

```diff
@@ -0,0 +1,173 @@
+---
+title: Using Machine ID With Github Actions
+description: A tutorial for using Machinie ID with Github Actions
+---
+
+Github actions are a popular CI/CD platform that works as a larger part of the Github
+ecosystem. Teleport, with the help of Machine ID, allows for Github actions to
+securely interact with Teleport Protected resources without the need for long lived
+credentials. By ensuring that rotated Machine ID credentials are additionally limited by RBAC,
+Github actions with Teleport can be applied as both a viable and secure solution whenever a
+CI/CD platform is needed.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A Linux based host that can support Machine ID. This "Machine ID Host
+  ```code
+  $ ssh -V
+  ```
+
+- A node that is a part of the Teleport cluster with the SSH service installed.
+
+- A machine with `tsh` access to the cluster.
+
+- A Github repository with Github actions enabled. This guide uses the example `gravitational/example`
+repo, however this value should be replaced with your own unique repo.
+
+## Step 1/3. Create a join Token for Github Actions
+
+Github Actions will need it's own join token in order to be able to join the cluster.
+Using a text editor of your choice, create a configuration YAML file `tokenconfig.yaml` for the
+join token that resembles the following:
+
+```yaml
+kind: token
+version: v2
+metadata:
+  name: github-demo
+  expires: "3000-01-01T00:00:00Z"
+spec:
+  roles: [Bot]
+  join_method: github
+  bot_name: github-demo
+  github:
+    allow:
+      - repository: gravitational/example
+```
+
+Let's go over the token configuration YAML in more detail:
+
+- `metadata.name` and `spec.bot_name` in this case represent the name of the
+bot that will be configured with Machine ID. In this case, the bot will be named
+`github-demo`, however this can be changed to any name of your choice. Note that this value will
+need to be used in other parts of the configuration later.
+
+- `metadata.expires` defines the date that the join token will expire. This example is set
+to the year `3000`, however care should be taken in production in order to Ensure
+that the expiration date will not be given a value that is greater than what is needed.
+
+- `spec.roles` defines the role for the bot user. The value of `[Bot]` ensures that only bots are
+allowed to join the cluster using this token, reducing the scope of privileges.
+
+- `spec.join_method` defines the join method the token is applicable for. Since this guide
+only focuses on Github Actions, this value should always be equal to `github`.
+
+- `spec.github.allow` field is used to set the Github repository that the Github actions will
```

Comment:
The allow block actually has several different options other than `repository` and I think we should either list them on this page, or in a reference elsewhere.

```
    // Sub also known as Subject is a string that roughly uniquely indentifies
    // the workload. The format of this varies depending on the type of
    // github action run.
    string Sub = 1 [(gogoproto.jsontag) = "sub,omitempty"];
    // The repository from where the workflow is running.
    // This includes the name of the owner e.g `gravitational/teleport`
    string Repository = 2 [(gogoproto.jsontag) = "repository,omitempty"];
    // The name of the organization in which the repository is stored.
    string RepositoryOwner = 3 [(gogoproto.jsontag) = "repository_owner,omitempty"];
    // The name of the workflow.
    string Workflow = 4 [(gogoproto.jsontag) = "workflow,omitempty"];
    // The name of the environment used by the job.
    string Environment = 5 [(gogoproto.jsontag) = "environment,omitempty"];
    // The personal account that initiated the workflow run.
    string Actor = 6 [(gogoproto.jsontag) = "actor,omitempty"];
    // The git ref that triggered the workflow run.
    string Ref = 7 [(gogoproto.jsontag) = "ref,omitempty"];
    // The type of ref, for example: "branch".
    string RefType = 8 [(gogoproto.jsontag) = "ref_type,omitempty"];
```

Hence I would modify the description to state "`spec.github.allow` is used to set rules for what GitHub Actions will be able to authenticate using this token. In our example, `repository` is set to only allow GitHub Actions running within that repository to authenticate. A full list of options can be found at XYZ"